src/dapps/selectors.ts

```diff
@@ -10,3 +17,39 @@ export const activeDappSelector = (state: RootState) =>
   state.dapps.dappsWebViewEnabled ? state.dapps.activeDapp : null
 
 export const dappsWebViewEnabledSelector = (state: RootState) => state.dapps.dappsWebViewEnabled
+
+export const dappsCategoriesSelector = (state: RootState) => state.dapps.dappsCategories
+
+export const dappsListSelector = (state: RootState) => state.dapps.dappsList
+
+export const dappsListLoadingSelector = (state: RootState) => state.dapps.dappsListLoading
+
+export const dappsListErrorSelector = (state: RootState) => state.dapps.dappsListError
+
+export const featuredDappSelector = createSelector(dappsListSelector, (dapps) => {
+  return dapps.find((dapp) => dapp.isFeatured)
+})
+
+export const dappCategoriesByIdSelector = createSelector(
+  dappsListSelector,
+  dappsCategoriesSelector,
+  (dapps, categories) => {
+    const mappedCategories: {
+      [id: string]: CategoryWithDapps
```

Comment:
For correctness:
```suggestion
      [id: string]: CategoryWithDapps | undefined
```

Reply:
done

Reply:
although this correctness means we now have a typeguard, to get around this error https://github.com/valora-inc/wallet/runs/7025166914?check_suite_focus=true - it looks a little weird